clang-tools-extra/test/clang-tidy/checkers/bugprone/unchecked-optional-access.cpp

```diff
@@ -180,3 +180,44 @@ void std_forward_rvalue_ref_safe(absl::optional<int>&& opt) {
 
   std::forward<absl::optional<int>>(opt).value();
 }
+
+namespace std {
+
+template<typename T>
+class optional {
+public:
+	template <typename U>
+  optional& operator=(const U &u){
+		data = u;
+	}
+private:
+  T data;
+};
+
+template <typename T>
+class vector {
+public:
+  vector() {}
+  T &operator[](unsigned long index) { return data[index]; }
+  bool empty() { return size == 0UL; }
+
+private:
+  T *data;
+  unsigned long size;
```

Comment:
I think simpler would be better here -- please leave out the definitions of the methods and the private data.